README.md

```diff
@@ -1 +1,157 @@
+features, while continuing to provide a stable science environment to the user. For example, a security update of the
+Jupyter environment can be made on top of the current latest science image without modifying it in any way.
+
+# How it works
+
+## Creating images
+
+Each science image is defined in the typical way, and the Dockerfiles for them need not contain any special instructions
+or any SciServer related items. While these images can be derivitive from eachother, there is no such requirement. Each
+change to contents of a science build dirctory (those directories located under `science/`) will, in principal, result
+in a build science image which is uniquely tagged. The top-level such science image of a particular image-version (see
+more below) will further result in a compatibility (compat) build using the compatibility layer defined in its image definition
+(`sciserver-image.json` file).
+
+The compat build generates a Dockerfile originated from the appropriate science image, and concatenates all dependent
+compat layers (those listed as "includes" in its `sciserver-image.json` file) to result in the final Dockerfile and
+build context directory. Any files existing more than one compat layer will be overwritten by the top-most layer (as is
+the case for a standard docker build - we don't use the docker build mechanism here as there is only one sensible image
+that can result from a compat layer build).
+
+All of the build products are uploaded (if specified) to a remote repository for consumption and will be tagged both
+with the version (akin to latest) and version plus build info (never changes):
+
+* science layer under `{repo}/sci/{target}` 
+* compat layer under `{repo}/rel/{image}`
+
+Future updates can reference one of the science layer products to make an updated without resulting in a new build (and
+potentially new versions of downloaded packages). Any time a compat layer is changed, new images will be created from
+the top-level science images for each image-version.
+
+## Consuming in SciServer compute
+
+Compute should always refer to a plain version-tagged compat image. Many such versions can exist for a single image, but
+they should always point to the latest compat image for that image-version. For example, if a build resulting in a final
+compat image for sciserver-essentials 2.4 in commit dk8912nds, there will be
+`{repo}/rel/sciserver-essentials:2.4-32-dk8912nds` (or similar) and `{repo}/rel/sciserver-essentials:2.4`, which have
+the same contents. In principal, compute should point to `{repo}/rel/sciserver-essentials:2.4` and always pull the
+image, so that compat updates can be deployed even when there are no underlying science changes.
+
+# Setting up images
+
+Each science and compat directory should have a `sciserver-image.json` file describing details about the image and how
+to create it. For **science** it looks like:
+
+```json
+{
+  # image is the is the common name of the resultant image
```

Comment:
This comment doesn't make any sense. Could you revise it?

Reply:
updated, I hope the new one provides a better description. Thanks!

Reply:
Yes, thanks!